controllers/component_build_controller.go

```diff
@@ -718,6 +721,80 @@ func GeneratePipelineRun(component appstudiov1alpha1.Component, bundle string, p
 	return yamlformat, nil
 }
 
+// TODO return additional pipeline params as well? Or merge with GeneratePipelineRun ?
+func (r *ComponentBuildReconciler) GetPipelineForComponent(ctx context.Context, component *appstudiov1alpha1.Component) (*tektonapi.PipelineRef, error) {
+	var pipelineSelectors []buildappstudiov1alpha1.InitialBuildPipelineSelector
+	pipelineSelector := &buildappstudiov1alpha1.InitialBuildPipelineSelector{}
+
+	globalPipelineSelectorRulesKey := types.NamespacedName{Namespace: buildServiceNamespaceName, Name: buildPipelineSelectorResourceName}
+	pipelineSelectorKeys := []types.NamespacedName{
+		// First try specific config for the application
+		{Namespace: component.Namespace, Name: component.Spec.Application},
+		// Second try namespaced config
+		{Namespace: component.Namespace, Name: buildPipelineSelectorResourceName},
+		// Finally try global config
+		globalPipelineSelectorRulesKey,
+	}
+
+	for _, pipelineSelectorKey := range pipelineSelectorKeys {
+		if err := r.Client.Get(ctx, pipelineSelectorKey, pipelineSelector); err != nil {
+			if !errors.IsNotFound(err) {
+				return &tektonapi.PipelineRef{}, err
+			}
+			// The config is not found, try the next one in the hierarchy
+		} else {
+			pipelineSelectors = append(pipelineSelectors, *pipelineSelector)
+		}
+	}
+
+	if len(pipelineSelectors) == 0 {
+		// Initial build pipeline selection logic is not defined.
+		// Create default configuration.
```

Comment:
do we need to handle this? if we just skip the if then it will end-up with deafult fallback bundle (which would happen also in this case)

Reply:
@Michkov, I thought that it would be nice to have explicit configuration in place, so admin can just edit it instead of searching across the docs and creating the object manually. I'm not insist on it, though. So, if you think that it's redundant, I'll delete it.

Reply:
It does seem odd to me to have a global object lazily created as part of a reconcile loop. If you were going to auto create this wouldn't it make more sense to do it at startup?

Reply:
I'm not sure if we need to create it at all... What do you thinks?